### PR TITLE
Clipping bbox in the mAP evaluator calculation.

### PR DESCRIPTION
**Pull request by qingqing01**
Fix https://github.com/PaddlePaddle/Paddle/issues/8871

### CODE REVIEW

File: paddle/fluid/operators/detection_map_op.h [hunk 1]
```diff
@@ -144,6 +144,15 @@ class DetectionMAPOpKernel : public framework::OpKernel<T> {
     }
   }
 
+  inline void ClipBBox(const Box& bbox, Box* clip_bbox) const {
+    T one = static_cast<T>(1.0);
+    T zero = static_cast<T>(0.0);
+    clip_bbox->xmin = std::max(std::min(bbox.xmin, one), zero);
```
Comment (pkuyym) [inline paddle/fluid/operators/detection_map_op.h]: clip_bbox --> clipped_bbox

File: paddle/fluid/operators/detection_map_op.h [hunk 2]
```diff
@@ -144,6 +144,15 @@ class DetectionMAPOpKernel : public framework::OpKernel<T> {
     }
   }
 
+  inline void ClipBBox(const Box& bbox, Box* clip_bbox) const {
```
Comment (pkuyym) [inline paddle/fluid/operators/detection_map_op.h]: Or you can pass `Box& clipped_bbox`

Comment (qingqing01) [reply paddle/fluid/operators/detection_map_op.h]: I do not change it. Thanks!

File: paddle/fluid/operators/detection_map_op.h (hunk 1 above)
Comment (qingqing01) [reply paddle/fluid/operators/detection_map_op.h]: Done. Thanks!

**Pull request merged**